design-system/src/theme-provider.tsx

```diff
@@ -92,17 +92,20 @@ ThemeProvider.defaultProps = {
   theme: 'default',
 };
 
-const useTheme = (parentSelector = defaultParentSelector) => {
-  const [theme, setTheme] = useState<string>('default');
+const useTheme = (parentSelector = defaultParentSelector): ThemeName => {
```

Comment:
I'm starting to think that current return type of this hook (a `string`) is too restrictive. If we need to return something else from the hook in the future, consumers would be affected.
Perhaps it would be better to return and object with the selected theme as a property so it would be easier to add something in the future (I'm thinking in an imperative function to update the theme, for instance).

What do you think?

Reply:
I guess it depends how we envision the usage of it. Also, I'm wondering about the `useEffect` implementation, as it seems it only runs once but what about when the theme changes due to other factors?

Maybe it's too soon to think about it (for the redesign work) and it's more interesting when we might have multiple themes to choose from.

Reply:
I think having a string as a return type is too restricting. I don't know how this will evolve in the future, but having an object as return type would make potential changes easier to digest for consumers.

Regarding the effect, yes it was only running once as we didn't have requirements to have it in sync. But now I already found an issue with this (when we need to re-render a component using `useTheme`, for example the `Tag` component now).
I already have an idea about how to tackle that issue but I prefer to open a separate PR for that.